### DISCUSSION TITLE
Grid layout not working as expected

### DISCUSSION THREAD

**Discussion by 0x4a61636f62** [tailwindlabs/tailwindcss#12025] [Help] [answered]
I am trying to achieve a grid layout exactly like the following *HTML*:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Custom Grid</title>
  <style>
    .grid-container {
      display: grid;
      grid-template-columns: repeat(16, 1fr);
    }
    .grid-item-4 {
      grid-column: span 4;
    }
    .grid-item-12 {
      grid-column: span 12;
    }
  </style>
</head>
<body>
  <div class="grid-container">
    <section class="grid-item-4">
      <h1>Left</h1>
    </section>
    <article class="grid-item-12">
      <h1>Right</h1>
    </article>
  </div>
</body>
</html>
```

And I expected the following *Tailwind* example to produce the same result:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Custom Grid</title>
  <!-- Make sure to include Tailwind CSS in your project -->
  <link href="https://cdn.jsdelivr.net/npm/tailwindcss@2.2.15/dist/tailwind.min.css" rel="stylesheet">
</head>
<body>
  <div class="grid grid-cols-16">
    <section class="col-span-4">
      <h1 class="text-xl font-bold">Left</h1>
    </section>
    <article class="col-span-12">
      <h1 class="text-xl font-bold">Right</h1>
    </article>
  </div>
</body>
</html>
```

However, in the Tailwind example, the `<section>` and `<article>` elements are stacked vertically instead of horizontally. It can be reproduced in, e.g., `codepen.io`.

What am I missing?

**Comment by wongjn** [comment, marked as answer]
You are missing the value for `grid-cols-16` to presumably resolve to `grid-template-columns: repeat(16, 1fr)`. As per [the documentation](https://tailwindcss.com/docs/grid-template-columns),  `grid-cols-*` values only go up to 12 with default configuration. [See the documentation for adding custom values](https://tailwindcss.com/docs/grid-template-columns#using-custom-values).

**Reply by 0x4a61636f62** [reply]
Thank you @wongjn, I thought that the value was arbitrary.

**Reply by zkmark** [reply]
I think the documentation is wrong, it should start with **span** right?

```
<div class="grid grid-cols-16">
	<!-- Error docs
	<div class="col-[16_/_span_16]">
	</div>
		-->
	<!-- Ok -->
	<div class="col-[span_16/_span_16]">span 16/ span 16</div>
</div>
```

**Comment by rifazul123** [comment]
We sometimes forget about the default configuration of any layout like grid layout, flex layout.
